relay-server/src/actors/project_upstream.rs

```diff
@@ -28,6 +28,15 @@ pub struct GetProjectStates {
     pub full_config: bool,
 }
 
+impl GetProjectStates {
+    pub fn is_full_config(&self) -> bool {
+        #[cfg(feature = "processing")]
+        return self.full_config;
```

Comment:
I think we should always return the value of `full_config` here, since this is then validated based on the status of the requesting relay. It's the requesting Relay that asks for full config based on the processing flag.